### PR TITLE
deps: Update yaml_cpp deps to fix CRNL and emit issue.

### PR DESCRIPTION
**Pull request by vehre-x41**
Commit Message: deps: Update yaml_cpp deps to fix CRNL and emit issues.
Additional Description: Update reference to yaml_cpp to make use of latest fixes:
- fix where on Windows a CRLF is not detected correctly
- emit of keys having special chars, like &, lead to illegal yaml.
Signed-off-by: Andre Vehreschild <vehre@x41-dsec.de>
Risk Level: Low
Testing: n/a
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: Force yaml_cpp to link statically.

### CODE REVIEW

File: bazel/foreign_cc/yaml_cpp.patch [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
++})
++
+ cc_library(
+     name = "yaml-cpp_internal",
+     visibility = ["//:__subpackages__"],
+@@ -11,4 +17,5 @@ cc_library(
+     includes = ["include"],
+     hdrs = glob(["include/**/*.h"]),
+     srcs = glob(["src/**/*.cpp", "src/**/*.h"]),
++    defines = yaml_cpp_defines,
```
Comment (keith) [inline bazel/foreign_cc/yaml_cpp.patch:19]: Could this be in copts instead so these don't propagate up?

**Review by keith — COMMENTED** [review]
Are there upstream issues / PRs tracking these?

File: bazel/foreign_cc/yaml_cpp.patch (hunk 1 above)
Comment (vehre-x41) [reply bazel/foreign_cc/yaml_cpp.patch:19]: They unfortunately need to propagate up or build on windows fails due to the API of yaml_cpp. I tried the copts in the beginning, but never to succeeded.

**Comment by vehre-x41** [comment]
> Are there upstream issues / PRs tracking these?

Er, what do you mean? This work is motivated by [PR20544](https://github.com/envoyproxy/envoy/pull/20544) not being the 'right' solution. The issue is in the yaml_cpp library and at least partially fixed there. Unfortunately way my fix not perfect, so have to iterate another patch to the yaml_cpp lib. This PR here is just for testing, whether the patch of yaml_cpp now really fixes all issues and builds on all platforms.

**Comment by keith** [comment]
Sorry I just meant for the newline one, so https://github.com/jbeder/yaml-cpp/pull/1099, can you add that as a comment in the patch so we know we can remove that once it merges?

**Comment by vehre-x41** [comment]
/donotmerge

This is a temporary state for testing purpose only. DO NOT MERGE!

File: bazel/repository_locations.bzl [hunk 2]
```diff
@@ -557,14 +557,14 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         project_name = "yaml-cpp",
         project_desc = "YAML parser and emitter in C++ matching the YAML 1.2 spec",
         project_url = "https://github.com/jbeder/yaml-cpp",
-        version = "db6deedcd301754723065e0bbb1b75927c5b49c7",
-        sha256 = "387d7f25467312ca59068081f9a25bbab02bb6af32fd3e0aec1bd59163558171",
+        version = "fa744bbbe22da24c745d991fd8f6f0401af7acf7",
```
[the remark this answers is not in the document]
Comment (vehre-x41) [reply bazel/repository_locations.bzl]: yaml-cpp has merged #1100. Updated refs and now just waiting for CI to pass.

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

**Pull request merged**